src/main/java/io/cryostat/jmcagent/S3ProbeTemplateService.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.jmcagent;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import io.cryostat.ConfigProperties;
+import io.cryostat.Producers;
+import io.cryostat.StorageBuckets;
+import io.cryostat.core.FlightRecorderException;
+import io.cryostat.core.agent.ProbeTemplate;
+import io.cryostat.core.agent.ProbeTemplateService;
+
+import io.quarkus.runtime.StartupEvent;
+import io.smallrye.common.annotation.Blocking;
+import io.vertx.core.eventbus.EventBus;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hc.core5.http.ContentType;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.logging.Logger;
+import org.xml.sax.SAXException;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.DeleteObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectTaggingRequest;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Object;
+import software.amazon.awssdk.services.s3.model.Tag;
+import software.amazon.awssdk.services.s3.model.Tagging;
+
+public class S3ProbeTemplateService implements ProbeTemplateService {
+
+    @ConfigProperty(name = ConfigProperties.AWS_BUCKET_NAME_EVENT_TEMPLATES)
+    String bucket;
+
+    @Inject S3Client storage;
+    @Inject StorageBuckets storageBuckets;
+
+    @Inject EventBus bus;
+
+    @Inject
+    @Named(Producers.BASE64_URL)
+    Base64 base64Url;
+
+    @Inject Logger logger;
+
+    void onStart(@Observes StartupEvent evt) {
+        storageBuckets.createIfNecessary(bucket);
+    }
+
+    @Override
+    @Blocking
```

Comment:
`@Blocking` should be removed from the methods within this utility class - it's only semantically meaningful for Quarkus when it's applied to "entrypoint" methods, ie ones that Quarkus itself will call into.